src/cargo/ops/cargo_install.rs

```diff
@@ -422,7 +423,7 @@ impl<'gctx> InstallablePackage<'gctx> {
         for &(bin, src) in binaries.iter() {
             let dst = staging_dir.path().join(bin);
             // Try to move if `target_dir` is transient.
-            if !self.source_id.is_path() && fs::rename(src, &dst).is_ok() {
+            if dry_run || (!self.source_id.is_path() && fs::rename(src, &dst).is_ok()) {
```

Comment:
Instead of this, should we just not execute the entire loop when `dry_run` is true?

Reply:
You're right, i fixed this!